docs/coil/migration-coilimage.md

```diff
@@ -0,0 +1,40 @@
+# Migration from CoilImage
+
+In Accompanist v0.8.0 the Coil library was refactored, moving away from the `CoilImage()` functions, to a new remembered painter provided by [`rememberCoilPainter()`][rememberpainter]. This page details how to migrate, as well as alternatives for function which no longer exists.
+
+## CoilImage is now deprecated
+The `CoilImage()` functions still exist (as of v0.8.0) but are now deprecated. They will be removed in a future release of Accompanist (likely v1.0).
+
+### ReplaceWith
+
+An automatic replacement migration has been provided in the deprecation. This allows you to migrate one or all `CoilImage()` calls over to the new API.  This works perfectly in most cases but you may find that the resulting code needs tidying up, YMMV.
+
+## Removed features
+
+The new API is intentionally missing some functionality which was previously available in `CoilImage`. Migrating usages of those features requires manual migration:
+
+### onRequestCompleted()
+
+`CoilImage` previously allowed the passing of a lambda which was invoked whenever a request was completed. This has been removed to simplify the internal state. If you wish to convert state changes into events, you can use [`snapshotFlow()`][snapshotflow] to observe changes to `painter.loadState`, and then call your logic as necessary:
```

Comment:
Should this be in the main coil doc, not just in the migration guide? Imagine this page will go away at some point.

Reply:
Done